recipe/meta.yaml

```diff
@@ -1,31 +1,32 @@
 {% set name = "ctypesgen" %}
-{% set version = "1.1.2.dev4+g8a62d02" %}
+{% set version = "1.1.1a" %}
 
 package:
-  name: {{ name|lower }}
+  name: {{ name|lower }}-pypdfium2-team
   version: {{ version }}
 
 source:
-  url: file:///Users/jolivieri/repos/upstream/ctypesgen-pypdfium2-team/dist/ctypesgen-1.1.2.dev4+g8a62d02.tar.gz
-  sha256: d23910fd783b8eb89d30cbb51b9c1bbedcc4143a50617c69959da4951cac94f9
+  git_url: https://github.com/pypdfium2-team/{{ name }}.git
+  git_rev: 8a62d02a6fd350df6d22e179bcd164df0da770d4
 
 build:
   entry_points:
     - ctypesgen = ctypesgen.main:main
-  noarch: python
-  script: {{ PYTHON }} -m pip install . -vv
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
   number: 0
 
 requirements:
+  build:
+    - git  # [not win]
   host:
-    - python >=3.7
+    - python
     - setuptools >=44
     - wheel
     - setuptools-scm >=3.4.3
     - toml
     - pip
   run:
-    - python >=3.7
+    - python
```

Comment:
We should consider adding [a `run_constrained` section](https://docs.conda.io/projects/conda-build/en/stable/resources/define-metadata.html#run-constrained`) to prevent this and the [upstream `ctypesgen` package](https://github.com/AnacondaRecipes/ctypesgen-feedstock) from being installed in the same environment. Something like:
```
run_constrained:
  - ctypesgen <=0.0.0dev0
```

(conda doesn't really have the concept of a `conflicts`-type specification, so we have to use a specification like `<=0.0.0dev0` as a substitute. This `run_constrained` tells the conda solver that if `ctypesgen-pypdfium2-team` is to be installed into the same environment as the upstream `ctypesgen`, that upstream `ctypesgen` package must be less than version 0, which can't happen and block the install.)

Reply:
Interesting way of dealing with this issue. Do you think this is a feature worth implementing in `conda-build`?

Reply:
Also, is `dev0` really necessary?

Reply:
> Do you think this is a feature worth implementing in conda-build?
It would have to be in `conda` proper, and yes, it's something the community has considered adding in the past (along with optional dependencies). Mostly, we've been stuck on how to roll such features out while maintaining some semblance of compatibility with the long tail of users on older conda releases.

> Also, is dev0 really necessary?
Not necessary. I stuck `dev0` in there out of habit.